pkg/cac/cac.go

```diff
@@ -8,85 +8,88 @@ import (
 	"bytes"
 	"encoding/binary"
 	"errors"
+	"fmt"
 
 	"github.com/ethersphere/bee/pkg/bmtpool"
 	"github.com/ethersphere/bee/pkg/swarm"
 )
 
 var (
-	errTooShortChunkData = errors.New("short chunk data")
-	errTooLargeChunkData = errors.New("data too large")
+	ErrChunkDataShort = errors.New("chunk data length is not sufficient")
```

Comment:
Instead of two separate errors, I would suggest creating a new error type that will carry the invalid length:

```
type InvalidChunkDataLengthError uint

func (e InvalidChunkDataLengthError) Error() string {
	return fmt.Sprintf("invalid CAC data length: %d", uint(e))
}
```

and then just below return `InvalidChunkDataLengthError(dataLength)`.

Reply:
It looks that this case is more suitable for two errors, one error should indicate when **data** exceeds max allowed length, and second should indicate case when **span** size is not exactly `swarm.SpanSize`. There was problem with error text which has been corrected in new commits.

--- 

Just on the side note... Consider following case which uses `InvalidChunkDataLengthError` type as suggested:


```
func foo() error { // foo return error from `cac.New` and some other error
	err := foo1() 
	if err != nil {
		return err
	}
	
	return cac.New(...)
}

{ // some other logic call `foo`, and want's to know what has happend
	...
	err := foo()
	if error.Is(err, InvalidChunkDataLengthError(0)) { // but what if value could be 1, 2, 3, or 141 ...?
		// problem with cac
	} else {
		// some other problem
	}
	...
}

```

 Errors specified in suggested way implies that other logic which uses `cac` wouldn't be able to use `errors.Is(...)` when processing error result because it would need to provide exact value of data length which could be different in different cases. Therefore this construct should probably be avoided if code wants to be future-proof.

Reply:
The `errors.Is(...)` shouldn't be an issue, you can use `errors.As(err, new(InvalidChunkDataLengthError))` instead.